lib/ecto/adapters/sql.ex

```diff
@@ -991,19 +991,40 @@ defmodule Ecto.Adapters.SQL do
   defp log_stacktrace(nil), do: ""
 
   defp log_stacktrace(stacktrace) do 
-    {module, function, arity, metadata} = Enum.at(stacktrace, 2)
+    case first_non_ecto(stacktrace) do
+      {module, function, arity, metadata} ->
+        [
+          ?\n,
+          "↳ ",
+          inspect(module),
+          ?.,
+          Atom.to_string(function),
+          ?/,
+          inspect(arity),
+          ", at: ",
+          inspect(metadata)
+        ]
+      nil ->
+        ""
+    end
+  end
 
-    [
-      ?\n,
-      "↳ ",
-      inspect(module),
-      ?.,
-      Atom.to_string(function),
-      ?/,
-      inspect(arity),
-      ", at: ",
-      inspect(metadata)
-    ]
+  defp first_non_ecto(stacktrace) do
+    reversed = Enum.reverse(stacktrace)
+
+    reversed
+    |> Enum.find_index(fn {module, _function, _arity, _metadata} ->
```

Comment:
Do `Enum.drop_while` it doesn't start with "Elixir.Ecto". Then check if it is an empty list, if not, it is the first element.

Reply:
Can't do it this way because we may have something like `Enum` in the middle (its the case for `Ecto.Multi`)

Reply:
I can be any operation from Enum. The point is that we should be able to traverse once instead of doing it twice. :)

Reply:
Just this comment and we are good to go. Ping me if you are not sure how to go ahead!

Reply:
I got a very simple solution with recursion. But I want to experiment a last bit more today :)

Reply:
Instead of find_index maybe find_value? Drop_while would also work :)